logstash-plugins/usagetype-plugin/src/main/java/org/reactome/logstash/plugins/usagetype/util/IPUtils.java

```diff
@@ -0,0 +1,50 @@
+package org.reactome.logstash.plugins.usagetype.util;
+
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.regex.Pattern;
+
+public final class IPUtils
+{
+	private static final long MULT_FOR_FIRST_OCTET = 256 << 16;
+
+	private static final long MULT_FOR_SECOND_OCTET = 256 << 8;
```

Comment:
Just curious about this if you could explain how it works.  I've never really understood the bit-wise operator.

Reply:
`a << n` will shift the binary representation of `a` to the left by `n` positions.

A video:
https://www.youtube.com/watch?v=pv1C0_6k78A

Another page on converting IP addresses to decimal numbers:

https://mkyong.com/java/java-convert-ip-address-to-decimal-number/